File: development/subscription/tools/user-journey/src/test/scala/Subscriber.scala
import java.util.concurrent.TimeUnit

import com.affaince.subscription.date
import com.affaince.subscription.date.SysDate
import com.affaince.subscription.testdata.generator.{SubscriptionTestDataGenerator}
import io.gatling.core.Predef._
import io.gatling.core.session.el._
import io.gatling.core.structure.ChainBuilder
import io.gatling.http.Predef._
import org.joda.time.LocalDate

import scala.util.Random

/**
  * Created by rbsavaliya on 04-03-2016.
  */
class Subscriber extends BaseSimulator {

  var productTestDataGenerator = new SubscriptionTestDataGenerator().generate().getSubscriptionCount();

  if (productTestDataGenerator > 2000) {
    productTestDataGenerator = 2000;
  }

  val scn = scenario("Create Subscriber").exec(CreateSubscriber.createSubscriber)
    .pause(10)
    .repeat(1) {
      CreateSubscriber.setPassword
    }
    .repeat(1) {
      CreateSubscriber.createSubscription
    }
    .repeat(1) {
      CreateSubscriber.addItemToSubscription
   // }.pause(1)
    //.repeat(1) {
     // CreateSubscriber.sysDateChange
    }.pause(1)
      .repeat(1){
        CreateSubscriber.selectPaymentScheme
      }
    .repeat(1) {
      CreateSubscriber.confirmSubscription
    }

  val scenario2 = scenario("Add Delivery Charges").exec(SetDeliveryChargesRules.setDeliveryChargesRules)
  //setUp(scn.inject(rampUsers(productTestDataGenerator) over(productTestDataGenerator/15)).protocols(http), scenario2.inject(atOnceUsers(1)).protocols(http))

  setUp(scn.inject(atOnceUsers(1)).protocols(http), scenario2.inject(atOnceUsers(1)).protocols(http))


  //setUp(scn.inject(constantUsersPerSec(users.toDouble) during (duration.seconds))).protocols(http)
}

object CreateSubscriber {

  val createSubscriberUrl="http://localhost:8081/subscriber"
  val createSubscriptionUrl = "http://localhost:8081/subscription"
  val sysDateChangeUrl= "http://localhost:8086/sysdate"
  val subscriberJsonFeeder = jsonFile ("Subscribers.json")
  val getDeliveryIdsUrl = "http://localhost:8081/delivery/getDeliveryIds"
  val selectPaymentUrl = "http://localhost:8081/subscription/selectpayment"
 // val sysdateFeeder = csv("sysdate.csv").queue;

  val createSubscriber = feed(subscriberJsonFeeder)
    .exec(
      http ("Create Subscriber")
        .post(createSubscriberUrl)
        .body(
          StringBody(
            """
              |{
              |   "subscriberName": {"title":"${title}","firstName":"${firstName}",
              |    "middleName":"${middleName}",
              |    "lastName":"${lastName}"},
              |    "address":{"addressLine1":"${addressLine1}",
              |    "addressLine2":"${addressLine2}",
              |    "city":"${city}",
              |    "state":"${state}",
              |    "country":"${country}",
              |    "pinCode":"${pinCode}"},
              |    "contactDetails":{"email":"${email}",
              |    "mobileNumber":"${mobileNumber}",
              |    "alternativeNumber":"$alternativeNumber"}
              |}
            """.stripMargin
          )
        ).asJSON
        .check (jsonPath("$.id").saveAs("subscriberId"))
    )

  val setPassword = //exec(session => session.set("randomeName", Random.nextInt(5)))
    exec (
      http ("Set Subscriber Password")
        .put((createSubscriberUrl + "/password/${subscriberId}").el[String])
        .body(
          StringBody(
            """
              |{
              | "password":"password1"
              |}
            """.stripMargin
          )
        ).asJSON
    )

  val createSubscription =
    exec(
      http ("Create Subscription")
        .post(createSubscriptionUrl)
        .body(
          StringBody(
            """
              |{
              | "subscriberId":"${subscriberId}"
              |}
            """.stripMargin
          )
        ).asJSON
        .check(jsonPath("$.id").saveAs("subscriptionId"))
    )

  val addItemToSubscription =
    exec(
      http("Add Item To Subscription")
        .put((createSubscriptionUrl + "/addsubscription/${subscriberId}").el[String])
        .body(
          ElFileBody ("${subscriberId}.json")
        ).asJSON
    )

  val selectPaymentScheme =
    exec(
      http("Select Payment Scheme")
        .put((selectPaymentUrl + "/${subscriberId}").el[String])
        .body(
          ElFileBody("paymentscheme.json")
        ).asJSON
    )

  val confirmSubscription =
    exec(
      http("Confirm Subscription")
        .put((createSubscriptionUrl + "/confirmsubscription/${subscriberId}").el[String])
    )

  val getDeliveryIds =
    exec (
      http("getDeliveryIds")
        .get((getDeliveryIdsUrl + "/${subscriberId}/${subscriptionId}").el[String])
        .asJSON
    )
}

File: development/subscription/tools/user-journey/src/test/java/com/affaince/subscription/testdata/generator/Subscriber.java
package com.affaince.subscription.testdata.generator;

import com.affaince.subscription.common.vo.Address;
import com.affaince.subscription.common.vo.ContactDetails;
import com.affaince.subscription.common.vo.SubscriberName;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

/**
 * Created by rbsavaliya on 02-08-2015.
 */
public class Subscriber {
    private String title;
    private String firstName;
    private String middleName;
    private String lastName;
    private String addressLine1;
    private String addressLine2;
    private String city;
    private String state;
    private String country;
    private String pinCode;
    private String email;
    private String mobileNumber;
    private String alternativeNumber;

    public Subscriber(String title, String firstName, String middleName, String lastName, String addressLine1, String addressLine2, String city, String state, String country, String pinCode, String email, String mobileNumber, String alternativeNumber) {
        this.title = title;
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.addressLine1 = addressLine1;
        this.addressLine2 = addressLine2;
        this.city = city;
        this.state = state;
        this.country = country;
        this.pinCode = pinCode;
        this.email = email;
        this.mobileNumber = mobileNumber;
        this.alternativeNumber = alternativeNumber;
    }

    public String getTitle() {
        return title;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getAddressLine1() {
        return addressLine1;
    }

    public String getAddressLine2() {
        return addressLine2;
    }

    public String getCity() {
        return city;
    }

    public String getState() {
        return state;
    }

    public String getCountry() {
        return country;
    }

    public String getPinCode() {
        return pinCode;
    }

    public String getEmail() {
        return email;
    }

    public String getMobileNumber() {
        return mobileNumber;
    }

    public String getAlternativeNumber() {
        return alternativeNumber;
    }
}
